connectors/sources/salesforce.py

```diff
@@ -169,26 +206,118 @@ async def _select_queryable_fields(self, sobject, fields):
         queryable_fields = sobject_fields.get(sobject, [])
         return [f for f in fields if f.lower() in queryable_fields]
 
-    async def _get_non_bulk_query_pages(self, soql_query):
-        return await self.get_request(
-            QUERY_ENDPOINT,
+    async def _yield_non_bulk_query_pages(self, soql_query):
+        """loops through query response pages and yields lists of records"""
+        response = await self._get_json(
+            f"{self.base_url}{QUERY_ENDPOINT}",
             params={"q": soql_query},
         )
+        yield response.get("records")
+
+        if response.get("done") is False:
+            page_count = 1
+            next_url = response.get("nextRecordsUrl")
+            while True:
+                next_page = await self._get_json(next_url)
+                yield next_page.get("records")
+
+                page_count += 1
+                if (
+                    next_page.get("done") is True
+                    or page_count >= QUERY_PAGINATION_LIMIT
```

Comment:
I believe usage of `nextRecordsUrl` eliminates the need to have QUERY_PAGINATION_LIMIT? I thought it applies only to skip/take pagination

Reply:
I just didn't want an endless loop, in case _something_ went wrong. It's really just a safeguard.